<?xml version="1.0" ?>
<robot name="crane_x7">
  <link name="world"/>
  <link name="base_frame_link">
    <visual>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Base_Frame.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="almi_color">
        <color rgba="0.515625 0.52734375 0.53515625 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Base_Frame.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.00669 0.000049 0.017868"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.387313"/>
    </inertial>
  </link>
  <link name="base_link">
    <visual>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Base.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="red_color">
        <color rgba="0.8984375 0.0 0.0703125 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="0 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Base.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.00669 0.000049 0.017868"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.387313"/>
    </inertial>
  </link>
  <link name="LINK1">
    <visual>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link1.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="white_color">
        <color rgba="0.859375 0.859375 0.859375 1"/>
      </material>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link1.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.000013 -0.000407 0.046345"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.235377"/>
    </inertial>
  </link>
  <link name="LINK2">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="1.5708 0 0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link2.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="white_color"/>
    </visual>
    <collision>
      <origin rpy="1.5708 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link2.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="1.5708 0 0" xyz="0.000036 0.037855 -0.00018"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.120005"/>
    </inertial>
  </link>
  <link name="LINK2_COVER_L">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="-1.5708 0 0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link_Cover.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="red_color"/>
    </visual>
    <collision>
      <origin rpy="-1.5708 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link_Cover.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="-1.5708 0 0" xyz="0.000036 0.037855 -0.00018"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.120005"/>
    </inertial>
  </link>
  <link name="LINK2_COVER_R">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="1.5708 0 0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link_Cover.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="red_color"/>
    </visual>
    <collision>
      <origin rpy="1.5708 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link_Cover.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="1.5708 0 0" xyz="0.000036 0.037855 -0.00018"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.120005"/>
    </inertial>
  </link>
  <link name="LINK3">
    <visual>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link3.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="white_color"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link3.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="1.5708 0 0" xyz="-0.011533 0.000021 0.083707"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.340878"/>
    </inertial>
  </link>
  <link name="LINK4">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="1.5708 0 0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link4.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="white_color"/>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 0.0" rpy="1.5708 0 0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link4.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="1.5708 0 0" xyz="-0.008641 0.083071 -0.000086"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.230634"/>
    </inertial>
  </link>
  <link name="LINK4_COVER_L">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="-1.5708 0 0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link_Cover.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="red_color"/>
    </visual>
    <collision>
      <origin rpy="-1.5708 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link_Cover.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="-1.5708 0 0" xyz="0.000036 0.037855 -0.00018"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.120005"/>
    </inertial>
  </link>
  <link name="LINK4_COVER_R">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="1.5708 0 0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link_Cover.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="red_color"/>
    </visual>
    <collision>
      <origin rpy="1.5708 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link_Cover.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="1.5708 0 0" xyz="0.000036 0.037855 -0.00018"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.120005"/>
    </inertial>
  </link>
  <link name="LINK5">
    <visual>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link5.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="white_color"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link5.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="-0.000024 0.001285 0.063837"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.192709"/>
    </inertial>
  </link>
  <link name="LINK5_LOGO_RABBIT_L">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="-1.5708 1.5708 0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Logo_Rabbit.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="red_color"/>
    </visual>
    <collision>
      <origin rpy="-1.5708 1.5708 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Logo_Rabbit.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="-1.5708 1.5708 0" xyz="0.000036 0.037855 -0.00018"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.120005"/>
    </inertial>
  </link>
  <link name="LINK5_LOGO_CORP_L">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="-1.5708 1.5708 0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Logo_Corp.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="corp_color">
        <color rgba="0.2 0.2 0.2 1"/>
      </material>
    </visual>
    <collision>
      <origin rpy="-1.5708 1.5708 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Logo_Corp.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="-1.5708 1.5708 0" xyz="0.000036 0.037855 -0.00018"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.120005"/>
    </inertial>
  </link>
  <link name="LINK5_LOGO_RABBIT_R">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="1.5708 -1.5708 0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Logo_Rabbit.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="red_color"/>
    </visual>
    <collision>
      <origin rpy="1.5708 -1.5708 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Logo_Rabbit.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="1.5708 -1.5708 0" xyz="0.000036 0.037855 -0.00018"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.120005"/>
    </inertial>
  </link>
  <link name="LINK5_LOGO_CORP_R">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="1.5708 -1.5708 0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Logo_Corp.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="corp_color"/>
    </visual>
    <collision>
      <origin rpy="1.5708 -1.5708 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Logo_Corp.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="1.5708 -1.5708 0" xyz="0.000036 0.037855 -0.00018"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.120005"/>
    </inertial>
  </link>
  <link name="LINK6">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="1.5708 0 0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link6.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="white_color"/>
    </visual>
    <collision>
      <origin rpy="1.5708 0 0" xyz="0.0 0.0 0.0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link6.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="1.5708 0 0" xyz="0.005713 -0.003707 0.000781"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.130549"/>
    </inertial>
  </link>
  <link name="LINK7">
    <visual>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link7.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="red_color"/>
    </visual>
    <collision>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/Link7.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="0 0 0" xyz="0.000425 -0.000896 0.018691"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.119807"/>
    </inertial>
  </link>
  <link name="HAND_A">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="1.5708 0 0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/HandA.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="red_color"/>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 0.0" rpy="1.5708 0 0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/HandA.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="1.5708 0 0" xyz="-0.002987 0.030318 0.003073"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.015617"/>
    </inertial>
  </link>
  <link name="HAND_B">
    <visual>
      <origin xyz="0.0 0.0 0.0" rpy="1.5708 0 0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/HandB.stl" scale="1.0 1.0 1.0"/>
      </geometry>
      <material name="red_color"/>
    </visual>
    <collision>
      <origin xyz="0.0 0.0 0.0" rpy="1.5708 0 0"/>
      <geometry>
        <mesh filename="package://crane_x7_description/meshes/HandB.stl" scale="1.0 1.0 1.0"/>
      </geometry>
    </collision>
    <inertial>
      <origin rpy="1.5708 0 0" xyz="0.003058 0.031215 0.002548"/>
      <inertia ixx="1.0" ixy="0.0" ixz="0.0" iyx="0.0" iyy="1.0" iyz="0.0" izx="0.0" izy="0.0" izz="1.0"/>
      <mass value="0.014417"/>
    </inertial>
  </link>

<!--
  <gazebo>
    <plugin filename="libgazebo_ros_control.so" name="gazebo_ros_control">
      <robotNamespace>crane_x7</robotNamespace>
    </plugin>
  </gazebo>
-->

  <joint name="world_link1" type="fixed">
    <parent link="world" />
    <child link="base_link" />
    <origin xyz="0.0 0.0 0.005" rpy="0 0 0"/>
  </joint>
  <joint name="base_frame_link" type="fixed">
    <parent link="world" />
    <child link="base_frame_link" />
    <origin xyz="-0.015 0.0 0.005" rpy="0 0 0"/>
  </joint>
  <joint name="arm_joint1" type="revolute">
    <parent link="base_link" />
    <child link="LINK1" />
    <origin xyz="0.0 0.0 0.041" rpy="0 0 0"/>
    <limit lower="-2.96365" upper="2.97132" effort="0.5" velocity="4.81710873"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="arm_joint2" type="revolute">
    <parent link="LINK1" />
    <child link="LINK2" />
    <origin xyz="0.0 0.0 0.064" rpy="0 0 0"/>
    <limit lower="-1.60761" upper="1.59840" effort="0.5" velocity="4.81710873"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <joint name="arm_link2_cover_l" type="fixed">
    <parent link="LINK2" />
    <child link="LINK2_COVER_L" />
    <origin xyz="0.0 0.0316 0.0" rpy="0 0 0"/>
  </joint>
  <joint name="arm_link2_cover_r" type="fixed">
    <parent link="LINK2" />
    <child link="LINK2_COVER_R" />
    <origin xyz="0.0 -0.0316 0.0" rpy="0 0 0"/>
  </joint>
  <joint name="arm_joint3" type="revolute">
    <parent link="LINK2" />
    <child link="LINK3" />
    <origin xyz="0.0 0.0 0.065" rpy="0 0 0"/>
    <limit lower="-1.5708" upper="1.5708" effort="0.5" velocity="4.81710873"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="arm_joint4" type="revolute">
    <parent link="LINK3" />
    <child link="LINK4" />
    <origin xyz="0.0 0.0 0.185" rpy="0 0 0"/>
    <limit lower="-2.80718" upper="0.04141" effort="0.5" velocity="4.81710873"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <joint name="arm_link4_cover_l" type="fixed">
    <parent link="LINK4" />
    <child link="LINK4_COVER_L" />
    <origin xyz="0.0 0.025 0.0" rpy="0 0 0"/>
  </joint>
  <joint name="arm_link4_cover_r" type="fixed">
    <parent link="LINK4" />
    <child link="LINK4_COVER_R" />
    <origin xyz="0.0 -0.025 0.0" rpy="0 0 0"/>
  </joint>
  <joint name="arm_joint5" type="revolute">
    <parent link="LINK4" />
    <child link="LINK5" />
    <origin xyz="0.0 0.0 0.121" rpy="0 0 0"/>
    <limit lower="-2.77343" upper="1.51403" effort="0.5" velocity="4.81710873"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="arm_link5_logo_rabbit_l" type="fixed">
    <parent link="LINK5" />
    <child link="LINK5_LOGO_RABBIT_L" />
    <origin xyz="0.0 0.0345 0.132" rpy="0 0 0"/>
  </joint>
  <joint name="arm_link5_logo_corp_l" type="fixed">
    <parent link="LINK5" />
    <child link="LINK5_LOGO_CORP_L" />
    <origin xyz="0.0 0.0345 0.095" rpy="0 0 0"/>
  </joint>
  <joint name="arm_link5_logo_rabbit_r" type="fixed">
    <parent link="LINK5" />
    <child link="LINK5_LOGO_RABBIT_R" />
    <origin xyz="0.0 -0.0345 0.065" rpy="0 0 0"/>
  </joint>
  <joint name="arm_link5_logo_corp_r" type="fixed">
    <parent link="LINK5" />
    <child link="LINK5_LOGO_CORP_R" />
    <origin xyz="0.0 -0.0345 0.104" rpy="0 0 0"/>
  </joint>
  <joint name="arm_joint6" type="revolute">
    <parent link="LINK5" />
    <child link="LINK6" />
    <origin xyz="0.0 0.0 0.129" rpy="0 0 0"/>
    <limit lower="-1.57079" upper="1.57233" effort="0.5" velocity="4.81710873"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <joint name="arm_joint7" type="revolute">
    <parent link="LINK6" />
    <child link="LINK7" />
    <origin xyz="0.0 0.0 0.019" rpy="0 0 0"/>
    <limit lower="-2.96365" upper="2.97132" effort="0.5" velocity="4.81710873"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="hand_joint1" type="revolute">
    <parent link="LINK7" />
    <child link="HAND_A" />
    <origin xyz="-0.012 0.0 0.024" rpy="0 0 0"/>
    <limit lower="0.00460" upper="1.62601" effort="0.5" velocity="4.81710873"/>
    <axis xyz="0 -1 0"/>
  </joint>
  <joint name="hand_joint1_dummy" type="revolute">
    <parent link="LINK7" />
    <child link="HAND_B" />
    <origin xyz="0.012 0.0 0.024" rpy="0 0 0"/>
    <limit lower="0.00460" upper="1.62601" effort="0.5" velocity="4.81710873"/>
    <axis xyz="0 -1 0"/>
	<mimic joint="hand_joint1" multiplier="-1" offset="0"/>
  </joint>

  <transmission name="arm_joint1_trans">
    <provideFeedback>true</provideFeedback>
    <implicitSpringDamper>1</implicitSpringDamper>
    <motorTorqueConstant>0.418</motorTorqueConstant>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="arm_joint1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="arm_joint2_trans">
    <provideFeedback>true</provideFeedback>
    <implicitSpringDamper>1</implicitSpringDamper>
    <motorTorqueConstant>0.418</motorTorqueConstant>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="arm_joint2">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="3">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="arm_joint3_trans">
    <provideFeedback>true</provideFeedback>
    <implicitSpringDamper>1</implicitSpringDamper>
    <motorTorqueConstant>0.418</motorTorqueConstant>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="arm_joint3">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="4">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="arm_joint4_trans">
    <provideFeedback>true</provideFeedback>
    <implicitSpringDamper>1</implicitSpringDamper>
    <motorTorqueConstant>0.418</motorTorqueConstant>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="arm_joint4">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="5">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="arm_joint5_trans">
    <provideFeedback>true</provideFeedback>
    <implicitSpringDamper>1</implicitSpringDamper>
    <motorTorqueConstant>0.418</motorTorqueConstant>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="arm_joint5">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="6">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="arm_joint6_trans">
    <provideFeedback>true</provideFeedback>
    <implicitSpringDamper>1</implicitSpringDamper>
    <motorTorqueConstant>0.418</motorTorqueConstant>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="arm_joint6">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="7">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="arm_joint7_trans">
    <provideFeedback>true</provideFeedback>
    <implicitSpringDamper>1</implicitSpringDamper>
    <motorTorqueConstant>0.418</motorTorqueConstant>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="arm_joint7">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="8">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <transmission name="hand_joint1_trans">
    <provideFeedback>true</provideFeedback>
    <implicitSpringDamper>1</implicitSpringDamper>
    <motorTorqueConstant>0.418</motorTorqueConstant>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="hand_joint1">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
    </joint>
    <actuator name="9">
      <hardwareInterface>hardware_interface/PositionJointInterface</hardwareInterface>
      <mechanicalReduction>1</mechanicalReduction>
    </actuator>
  </transmission>
  <!--transmission name="hand_joint1_dummy_trans">
    <provideFeedback>true</provideFeedback>
    <implicitSpringDamper>1</implicitSpringDamper>
    <motorTorqueConstant>0.418</motorTorqueConstant>
    <type>transmission_interface/SimpleTransmission</type>
    <joint name="hand_joint1_dummy">
      <hardwareInterface>PositionJointInterface</hardwareInterface>
    </joint>
  </transmission-->

  <gazebo reference="base_link">
    <material>Gazebo/Green</material>
    <dampingFactor>0.5</dampingFactor>
    <selfCollide>false</selfCollide>
    <mu1>0.3</mu1>
    <mu2>0.3</mu2>
  </gazebo>
  <gazebo reference="LINK1">
    <material>Gazebo/Green</material>
    <dampingFactor>0.5</dampingFactor>
    <selfCollide>false</selfCollide>
    <mu1>0.3</mu1>
    <mu2>0.3</mu2>
  </gazebo>
  <gazebo reference="LINK2">
    <material>Gazebo/Green</material>
    <dampingFactor>0.5</dampingFactor>
    <selfCollide>false</selfCollide>
    <mu1>0.3</mu1>
    <mu2>0.3</mu2>
  </gazebo>
  <gazebo reference="LINK3">
    <material>Gazebo/Green</material>
    <dampingFactor>0.5</dampingFactor>
    <selfCollide>false</selfCollide>
    <mu1>0.3</mu1>
    <mu2>0.3</mu2>
  </gazebo>
  <gazebo reference="LINK4">
    <material>Gazebo/Green</material>
    <dampingFactor>0.5</dampingFactor>
    <selfCollide>false</selfCollide>
    <mu1>0.3</mu1>
    <mu2>0.3</mu2>
  </gazebo>
  <gazebo reference="LINK5">
    <material>Gazebo/Green</material>
    <dampingFactor>0.5</dampingFactor>
    <selfCollide>false</selfCollide>
    <mu1>0.3</mu1>
    <mu2>0.3</mu2>
  </gazebo>
  <gazebo reference="LINK6">
    <material>Gazebo/Green</material>
    <dampingFactor>0.5</dampingFactor>
    <selfCollide>false</selfCollide>
    <mu1>0.3</mu1>
    <mu2>0.3</mu2>
  </gazebo>
  <gazebo reference="LINK7">
    <material>Gazebo/Green</material>
    <dampingFactor>0.5</dampingFactor>
    <selfCollide>false</selfCollide>
    <mu1>0.3</mu1>
    <mu2>0.3</mu2>
  </gazebo>
  <gazebo reference="HAND_A">
    <material>Gazebo/Green</material>
    <dampingFactor>0.5</dampingFactor>
    <selfCollide>false</selfCollide>
    <mu1>0.3</mu1>
    <mu2>0.3</mu2>
  </gazebo>
  <!--gazebo reference="HAND_B">
    <material>Gazebo/Green</material>
    <dampingFactor>0.5</dampingFactor>
    <selfCollide>false</selfCollide>
    <mu1>0.3</mu1>
    <mu2>0.3</mu2>
  </gazebo-->
</robot>
